cpp/src/parquet/page_index.h

```diff
@@ -332,7 +340,8 @@ class PARQUET_EXPORT OffsetIndexBuilder {
 class PARQUET_EXPORT PageIndexBuilder {
  public:
   /// \brief API convenience to create a PageIndexBuilder.
-  static std::unique_ptr<PageIndexBuilder> Make(const SchemaDescriptor* schema);
+  static std::unique_ptr<PageIndexBuilder> Make(const SchemaDescriptor* schema,
+                                                InternalFileEncryptor* file_encryptor);
```

Comment:
These APIs are probably not widely-used, but they are still public and we should minimize API compatibility breakage. Can we make the encryption/decryption arguments optional? (or have overloads without them)

Reply:
OK, I prefer the latter which does not have a default parameter.